lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4020,7 +4027,7 @@ namespace {
 
       FuncDecl *setter = importAccessor(clangSetter,
                                         original->getDeclContext());
-      if (!setter)
+      if (!setter || !isExpectedTypeForPropertySetter(setter, original))
```

Comment:
When might this second part happen? Preferably a comment here at the use site, or else a comment on the function itself. Otherwise, it's hard to quickly think about when this check might trigger or why it is in here.